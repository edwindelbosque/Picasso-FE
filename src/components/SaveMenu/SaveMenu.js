import React from 'react';
import './SaveMenu.scss';

const SaveMenu = ({ catalogs, closeSaveMenu, showSaveMenu, postPalette }) => {
	const catalogList = catalogs.map(catalog => {
		return (
			<li onClick={() => savePalette(catalog.id)}>{catalog.catalogName}</li>
		);
	});

	const savePalette = id => {
		postPalette(id);
		closeSaveMenu();
	};

	return (
		<div className={`SaveMenu ${showSaveMenu ? 'showSaveMenu' : ''}`}>
			<h3>Catalogs</h3>
			<ul>{catalogList}</ul>
			<h4>Or create a new catalog</h4>
			<input type='text' placeholder='New Catalog' />
			<button>Create & Save</button>
		</div>
	);
};

export default SaveMenu;
