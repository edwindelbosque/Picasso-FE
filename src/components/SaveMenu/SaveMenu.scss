.SaveMenu {
	position: fixed;
	right: -30vw;
	width: 250px;
	height: 500px;
	background: #2164f3;
	top: 150px;
	border-radius: 3px;
	padding: 20px;
	padding-top: 15px;
	display: flex;
	flex-direction: column;
	box-shadow: 0 3px 4px rgba(0, 0, 0, 0.144);
	transition: 0.25s ease-in-out;

	h3 {
		color: white;
		font-family: 'Roboto';
		font-size: 30px;
		width: 100%;
		text-align: center;
		padding-bottom: 0 px;
	}

	ul {
		margin: 30px 0px 25px 20px;
		border-top: 1px solid rgb(155, 189, 194);
		border-bottom: 1px solid rgb(161, 197, 202);
		padding-top: 5px;
		color: white;
		height: 180px;
		font-family: 'Roboto';
		padding-left: 0px;
		overflow-y: scroll;
	}

	li {
		padding: 2px 5px;
		border-radius: 2px;
		margin: 8px 0;
		font-size: 20px;
		cursor: pointer;

		&:hover {
			margin: 0px 0;
			display: inline-block;
			color: rgb(255, 145, 145);
			background: rgb(255, 255, 255);
		}
	}

	h4 {
		margin-top: 5px;
		color: white;
		width: 100%;
		font-size: 25px;
		text-align: center;
		margin-bottom: 20px;
	}

	input {
		height: 35px;
		border: none;
		border-radius: 3px;
		margin-bottom: 20px;
		padding-left: 8px;
		font-size: 16px;
		box-shadow: 0 2px 4px rgba(0, 0, 0, 0.068);

		&:focus {
			outline: none;
		}
	}

	button {
		height: 40px;
		border: none;
		cursor: pointer;
		border-radius: 3px;
		font-size: 16px;
		color: rgb(65, 65, 65);
		background: #f3c221;
		box-shadow: 0 2px 4px rgba(0, 0, 0, 0.158);
		transition: 0.15s;

		&:hover {
			background: #ffd54d;
		}

		&:focus {
			outline: none;
		}
	}
}

.showSaveMenu {
	right: 5px;
}
