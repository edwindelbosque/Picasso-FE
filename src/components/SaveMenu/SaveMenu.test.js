import React from 'react';
import SaveMenu from './SaveMenu';
import { shallow } from 'enzyme';
jest.mock('react-redux', () => ({
	useDispatch: () => {},
	useSelector: () => ({ catalogs: [{}, {}] })
}));

describe('SaveMenu', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<SaveMenu />);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
