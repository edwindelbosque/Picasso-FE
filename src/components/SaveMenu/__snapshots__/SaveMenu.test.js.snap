// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveMenu App should match snapshot 1`] = `
<div
  className="SaveMenu showSaveMenu"
>
  <h3>
    Catalogs
  </h3>
  <ul />
  <h4>
    Or create a new catalog
  </h4>
  <input
    onChange={[Function]}
    placeholder="New Catalog"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create & Save
  </button>
</div>
`;
